### PR TITLE
Added doctoc functionality to create TOC for markdown documents

### PR DESCRIPTION
**Pull request by m1yag1**
* Initialized npm for the repo
* Added doctoc as a dev dependency
* Added `node_modules` to .gitignore file
* Added npm script to package.json to run doctoc locally.
* Generated TOC for all documents =).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,14 @@
 # PySlackers Community
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
```
Comment (mattrasband) [inline README.md:3]: As weird as it sounds, do we need a CI process to run this and auto-commit/push on merge into master?

File: package.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "community",
+  "version": "0.0.1",
+  "description": "A collection of community specific resources, such as rules, COC, and team services",
+  "main": "index.js",
+  "scripts": {
+    "doctoc": "node_modules/.bin/doctoc ."
```
Comment (mattrasband) [inline package.json]: You shouldn't actually need the `node_modules/.bin` prefix for anything in the `scripts` section, NPM automatically adds that to the `PATH` when running `npm run <script_target>` (see [here](https://docs.npmjs.com/misc/scripts#path))

**Review by mattrasband — CHANGES_REQUESTED** [review]
Minor, and if you think it's nit picky feel free to ignore, but I would like to promote the best practices (I guess I am presuming the npm script path is considered as much)

File: package.json (hunk 2 above)
Comment (m1yag1) [reply package.json]: oh i didn't realize that. I'll fix 'er up.

File: README.md (hunk 1 above)
Comment (m1yag1) [reply README.md:3]: It doesn't sound weird at all. You're saying that once this is merged that travis runs the `npm run doctoc` command, right?

**Review by mattrasband — APPROVED** [review]
🌮

File: README.md [hunk 3]
```diff
@@ -45,3 +55,17 @@ are welcome to [join](https://pyslackers.com/slack/).
 
 If you have a GitHub account, remember to star this repository so you can
 have quick access to this documentation. If not, bookmark this page.
+
+# Creating the Table of Contents
```
Comment (ovv) [inline README.md:59]: Should this be in the `README.md` ? Maybe we should create a `CONTRIBUTING.md` :thinking:

File: README.md (hunk 1 above)
Comment (ovv) [reply README.md:3]: Having Travis run it would be awesome. I'm afraid it might drive people away it they need to figure out how to run `doctoc`.

**Comment by ovv** [comment]
@m1yag1

**Comment by m1yag1** [comment]
Closing PR as I probably won't have time to get back to this.

**Pull request closed without merging**